src/api/app/models/package.rb

```diff
@@ -839,6 +839,7 @@ def add_channels(mode=nil)
         next if mode == :skip_disabled and not cb.channel_binary_list.channel.is_active?
         cpkg = cb.create_channel_package_into(self.project)
         next unless cpkg
+        next if mode.nil? and not cb.channel_binary_list.channel.is_active?
```

Comment:
Can you use the "&& !" operators instead of "and not"?. The same for line 839...